geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitor.java

```diff
@@ -60,21 +60,24 @@ long getLastProcessCpuTime() {
     return lastProcessCpuTime;
   }
 
-  public VMStatsMonitor(String name) {
+  public VMStatsMonitor(String name, boolean processCPUTimeAvailable) {
     super(name);
-    processCPUTimeAvailable = MBeanJMXAdapter.isAttributeAvailable(PROCESS_CPU_TIME_ATTRIBUTE,
-        ManagementFactory.OPERATING_SYSTEM_MXBEAN_NAME);
+    this.processCPUTimeAvailable = processCPUTimeAvailable;
     if (!processCPUTimeAvailable) {
       cpuUsage = VALUE_NOT_AVAILABLE;
     }
   }
 
+  public VMStatsMonitor(String name) {
```

Comment:
Can we get rid of this constructor and just use the one above in the production code as well as the test code?  You could inject the `processCPUTimeAvailable` parameter in the production code as well using the above constructor.

Reply:
We cannot. The callers are in different packages.